### PR TITLE
Create BattleRoom class to reduce code duplication

### PR DESCRIPTION
**Pull request by JamesVanBoxtel**
• Created a BattleRoom class to handle data / methods that happen for a given set of battles (typically the length of a room, but this can be used outside net play)
• Move the background image globals to GAME
• Move the win counts to the battle room class, they will auto clear when a new room is started

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -31,6 +32,9 @@ local pointer_hidden = false
 
 -- Called at the beginning to load the game
 function love.load()
+  if os.getenv("LOCAL_LUA_DEBUGGER_VSCODE") == "1" then
```
Comment (Shosoul) [inline main.lua:35]: I don't think we should reference editors in the code when committing to code.

**Review by Shosoul — CHANGES_REQUESTED** [review]
remove editor reference

**Comment by Shosoul** [comment]
Also, maybe it should be part of the match class or something instead of it's own class

**Comment by Shosoul** [comment]
> Also, maybe it should be part of the match class or something instead of it's own class, I don't feel like it deserves to be it's own class with very limited functions that it provides.

Nevermind about this comment. We can keep it, but I still stand by removing the editor reference

**Comment by JamesVanBoxtel** [comment]
Yeah the idea of the "BattleRoom" is it will have all the VS stuff.
We can have a different class for handling puzzles and time trial and endless

**Review by Shosoul — APPROVED** [review]

**Pull request merged**